packages/client-admin/src/pages/projectsettings.tsx

```diff
@@ -0,0 +1,98 @@
+import { Box, Card, CardContent, CardHeader, FormLabel, IconButton } from '@mui/material';
+import { Formik, Form, Field } from 'formik';
+import { TextField, Icon as MuiIcon } from '@mui/material';
+import { useState } from 'react';
+import { Icon } from '@components/icon';
+import { faCaretDown } from '@fortawesome/free-solid-svg-icons';
+import { useGetProjectQuery } from '@graphql/project/project';
+import { useUpdateProjectMutation } from '@graphql/project/project';
+import { useAuth } from '../context/auth.context';
+
+const IconPreview = (props: any) => {
+  const { icon, size, imageSize } = props;
+
+  return (
+    <MuiIcon fontSize={size}>
+      <img src={icon} alt="icon" style={{ width: imageSize, height: imageSize }} />
+    </MuiIcon>
+  );
+};
+
+export const ProjectSettings = () => {
+  const { token, decoded_token, setToken } = useAuth();
+  const projectId = decoded_token?.projectId || '';
+  const [isDisabled, setIsDisabled] = useState(true);
+  const [updateProject] = useUpdateProjectMutation();
+  const { data: projectData, called, loading } = useGetProjectQuery({ variables: { id: projectId }, skip: !projectId });
+
+  const initialValues = {
+    id: projectData?.getProject.id || '',
+    name: projectData?.getProject.name || '',
+    description: projectData?.getProject.description || '',
+    logo: projectData?.getProject.logo || '',
+    homePage: projectData?.getProject.homePage || '',
+    redirectUrl: projectData?.getProject.redirectUrl || ''
+  };
+
+  return (
+    <Formik
+      initialValues={initialValues}
+      enableReinitialize={true}
+      onSubmit={async (values, { setSubmitting }) => {
+        try {
+          await updateProject({
+            variables: { id: values.id, name: values.name, description: values.description, logo: values.logo, homePage: values.homePage, redirectUrl: values.redirectUrl }
+          });
+
+          window.alert('Project Settings have been updated');
+        } catch (error) {
+          console.error(error);
+          window.alert('Error in updating project');
+        }
+
+        setSubmitting(false);
+        setIsDisabled(true);
+      }}
+    >
+      {({ isSubmitting, values }) => (
+        <Form>
+          <Card>
+            <CardHeader
+              action={
+                <IconButton disabled={isSubmitting} onClick={() => setIsDisabled(false)}>
+                  Edit
+                  <Icon icon={faCaretDown} />
+                </IconButton>
```

Comment:
Can't really have text in the Icon Button.
You can really un-edit, on click should more be a toggle

```suggestion
                <IconButton onClick={() => toggleEdit()}>
                  <Icon icon={isEditing ? faCircleXmark : faPen} />
                </IconButton>
```